packages/plugin-hardhat/src/utils/deploy.ts

```diff
@@ -9,7 +11,24 @@ export async function deploy(
   factory: ContractFactory,
   ...args: unknown[]
 ): Promise<Required<Deployment & DeployTransaction>> {
-  const { address, deployTransaction } = await factory.deploy(...args);
+  const contractInstance = await factory.deploy(...args);
+  const deployTransaction = contractInstance.deployTransaction;
+
+  const address: string = getContractAddress({
+    from: await factory.signer.getAddress(),
+    nonce: deployTransaction.nonce,
+  });
+  if (address !== contractInstance.address) {
+    debug(
+      'overriding contract address from ' +
```

Comment:
I prefer to write this as either `[...].join(' ')`, or template strings `` `...${addr}...` ``